### PR TITLE
fix(Select): Don't show first option if value = null

### PR DESCRIPTION
**Pull request by j-o-d-o**
PR to the discussion in #104. Need to figure out the desired behavior first. I will change this PR in case that changes to something other than showing the label / placeholder for value = null.

### CODE REVIEW

File: src/Select/index.js [hunk 1]
```diff
@@ -175,7 +175,10 @@ export const Select = withMDC({
       isFirstRun ||
       placeholderDidChange
     ) {
-      const newIndex = api.options.indexOf(api.nameditem(nextProps.value));
+      // if value = null, MDC will show the first option. This behaviour is not desired
+      let value = nextProps.value === null ? undefined : nextProps.value;
```
Comment (jamesmfriedman) [inline src/Select/index.js]: Use `const` when the value doesn't change.

**Comment by jamesmfriedman** [comment]
Just a heads up @j-o-d-o since you are a collaborator, you should be able to make branches directly inside of RMWC. When you do, just use the format yourname/your-fix.

Played around with this some, it definitely seems to be the desired behavior. I tested it against other falsey values and it worked just fine. I found out why in MDC this actually happens.

```javascript
function nameditem(key) {
  // NOTE: IE11 precludes us from using Array.prototype.find
  for (var i = 0, options = this.options, option; option = options[i]; i++) {
    if (option.id === key || option.getAttribute('name') === key) {
      return option;
    }
  }
  return null;
}
```

When option.id === key evaluates to false (the value wasn't found) it tries to fall back to option.getAttribute('name') which defaults to `null` when there is no "name" attribute on the DOM item. Definitely an odd quirk, but at least it makes some sense now.

Personally, I'd like to refactor this at some point to be completely value driven instead of using the namedItem api which looks at the DOM, but for now other than the comment to change `let` to `const`, It's fine to merge in and definitely fixes the issue.

**Comment by j-o-d-o** [comment]
Yeah, looks like that's it, good catch. Not sure if this is on purpose or not in mdc. I will check this out in the mdc chat and possibly create and issue in the mdc repo.

**Comment by jamesmfriedman** [comment]
I think it’s fine I’m the MDC code. Since mdc deals primarily with the DOM I would expect this kind of behavior.

**Pull request merged**